### PR TITLE
Optimize new tracing

### PR DESCRIPTION
**Pull request by jutaro**
Performance enhancements for new tracing (including  metrics, datapoints and forwarding)

### CODE REVIEW

**Review by fmaste — APPROVED** [review]

**Review by erikd — CHANGES_REQUESTED** [review]
IMO, that commit history badly needs to be squashed down.

File: cardano-node/cardano-node.cabal [hunk 1]
```diff
@@ -38,6 +38,7 @@ common project-config
                         -Wpartial-fields
                         -Wredundant-constraints
                         -Wunused-packages
+                        -fprof-auto
```
Comment (carbolymer) [inline cardano-node/cardano-node.cabal:41]: This introduces overhead in the runtime - do we want that to be merged into `master`?

**Review by carbolymer — CHANGES_REQUESTED** [review]
https://github.com/input-output-hk/cardano-node/pull/5403/files#r1274788872

File: cardano-node/cardano-node.cabal (hunk 1 above)
Comment (jutaro) [reply cardano-node/cardano-node.cabal:41]: I was not aware, that this actually gives a runtime overhead. I will remove it everywhere.

Comment (jutaro) [reply cardano-node/cardano-node.cabal]: Sorry, fixed it. The tracing tools are partly included in the node as libraries, so I care about them the same way

Comment (carbolymer) [reply cardano-node/cardano-node.cabal]: Thanks!

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs [hunk 2]
```diff
@@ -80,15 +80,18 @@ import qualified Ouroboros.Consensus.Protocol.Ledger.HotKey as HotKey
...
-                                          <> (Text.pack . show) remote)
+                                          <> forHuman remote)
     ]
+  forHuman (ConnectionId local' remote) =
+    "ConnectionId " <>  forHuman local' <> " " <> forHuman remote
+
 --------------------------------------------------------------------------------
 --   TraceLabelCreds peer a
---------------------------------------------------------------------------------
+--------------------------------------------------------------------------forHuman------
```
Comment (carbolymer) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs:94]: Accidental copy-paste?

**Review by carbolymer — APPROVED** [review]
approve for changes in `cardano-node`

**Review by disassembler — APPROVED** [review]

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs (hunk 2 above)
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs]: Fixed it

**Comment by disassembler** [comment]
LGTM, but can we clean up the commit history a bit before we put this into the merge queue? Thanks!

**Pull request merged**